src/sinks/prometheus/exporter.rs

```diff
@@ -374,28 +374,37 @@ fn handle(
     default_namespace: Option<&str>,
     buckets: &[f64],
     quantiles: &[f64],
-    metrics: &IndexMap<MetricRef, (Metric, MetricMetadata)>,
+    metrics: &RwLock<IndexMap<MetricRef, (Metric, MetricMetadata)>>,
     bytes_sent: &Registered<BytesSent>,
 ) -> Response<Body> {
     let mut response = Response::new(Body::empty());
 
-    if !authorized(&req, auth) {
-        *response.status_mut() = StatusCode::UNAUTHORIZED;
-        response.headers_mut().insert(
-            http::header::WWW_AUTHENTICATE,
-            HeaderValue::from_static("Basic, Bearer"),
-        );
-        return response;
-    }
+    match (authorized(&req, auth), req.method(), req.uri().path()) {
+        (false, _, _) => {
+            *response.status_mut() = StatusCode::UNAUTHORIZED;
+            response.headers_mut().insert(
+                http::header::WWW_AUTHENTICATE,
+                HeaderValue::from_static("Basic, Bearer"),
+            );
+        }
+
+        (true, &Method::GET, "/metrics") => {
+            let metrics = metrics.read().unwrap();
```

Comment:
`expect` instead?

Reply:
That was copied from the existing code, but I agree an expect would improve the situation.